src/modules/auth/token.service.ts

```diff
@@ -64,4 +68,26 @@ export class TokenService {
   async invalidateRefreshToken(id: string) {
     return this.refreshTokenRepository.deleteRefreshTokenById(id);
   }
+
+  async handleOriginCheck(req: Request, res: Response, next: NextFunction, service: JwtService) {
+    let token = null;
+    if (req.headers['authorization']) {
+      token = req.headers['authorization'].replace('Bearer ', '');
+    } else {
+      token = req.cookies.token
+    }
+
+    if (token) {
+      const decodedToken = service.decode(token) as TokenPayload;
+      const isBrowserRequestFromAuthenticatedOrigin = decodedToken.origin === req.headers['origin']
+      const isServerRequestOrGETFromSameDomain = req.headers['origin'] === undefined
+      if (isBrowserRequestFromAuthenticatedOrigin || isServerRequestOrGETFromSameDomain) {
+        next()
+      } else {
+        throw new InternalServerErrorException('Something went wrong')
+      }
+    } else {
+      throw new UnauthorizedException('Unauthorized')
```

Comment:
```suggestion
      throw new UnauthorizedException('Unauthorized');
```

Reply:
looks like there are instances where statements are terminated with semi-colon and some are not. let make them consistent.

Reply:
sure, thank you for suggestion